src/locales.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::HashMap;
+
+use fluent::{bundle::FluentBundle, FluentArgs, FluentResource, FluentValue};
+use intl_memoizer::concurrent::IntlLangMemoizer;
+use serde_json::Value;
+
+static FLUENT_EN: &str = include_str!("../locales/en.ftl");
+static FLUENT_ZH_CN: &str = include_str!("../locales/zh-CN.ftl");
+
+/// Fluent locale loader to localize text.
+///
+/// [`FluentLoader`] implements [`tera::Function`] trait,
+/// so it can be register as a tera function.
+pub struct FluentLoader {
+    bundle: FluentBundle<FluentResource, IntlLangMemoizer>,
+}
+
+impl FluentLoader {
+    pub fn new(locale: &str) -> Self {
+        let mut bundle =
+            FluentBundle::new_concurrent(vec![locale.parse().expect("Invalid locale string.")]);
+        if let Some(resource) = match locale {
+            "en" => FluentResource::try_new(FLUENT_EN.to_owned()).ok(),
+            "zh-CN" => FluentResource::try_new(FLUENT_ZH_CN.to_owned()).ok(),
+            _ => None,
```

Comment:
How about adding some warning message when the `locale` field is invalid.

Reply:
Sounds good.